### PR TITLE
Multiple where clauses implemented.

### PR DESCRIPTION
**Pull request by vqtran**
Multiple where clauses, check it out!

### CODE REVIEW

**Comment by vqtran** [comment]
Gah hold on, we can't have numbers in slot names apparently

**Comment by vqtran** [comment]
Fixed. There's some rough edges with multiple where's right now, for example it would be nice to say, "where product is either speakers or tv" and not "where product is speakers or product is tv".

**Comment by vqtran** [comment]
Switched our handling of Connection objects from static singletons to dependency injections - so now for testing purposes we can inject a connection to our test database and not airplaines.

File: speech-assets/build-utterances.js [hunk 1]
```diff
@@ -40,6 +40,9 @@ function buildTemplate(input, output) {
     // Expand each line out into its combinations.
     var expandedUtterances = '';
     data.split('\n').forEach(function(line) {
+      if (line == '' || line.charAt(0) == '#') {
+        return;
+      }
```
Comment (GabeLoins) [inline speech-assets/build-utterances.js]: comments!

**Comment by GabeLoins** [comment]
urge 2 merge

**Pull request merged**